internal/endorser/endorser.go

```diff
@@ -43,26 +44,35 @@ type ParsedProvenance struct {
 	SourceMetadata claims.ProvenanceData
 }
 
-// GenerateEndorsement generates an endorsement statement for the given validity duration, using
-// the given provenances as evidence and reference values to verify them. At least one provenance
-// must be provided. The endorsement statement is generated only if the provenance statements are
-// valid.
-func GenerateEndorsement(referenceValues *verification.ReferenceValues, validityDuration claims.ClaimValidity, provenances []ParsedProvenance) (*intoto.Statement, error) {
-	verifiedProvenances, err := verifyAndSummarizeProvenances(referenceValues, provenances)
+// GenerateEndorsement generates an endorsement statement for the given binary
+// and the given validity duration, using the given provenances as evidence and
+// VerificationOptions to verify them. If more than one provenance statements
+// are provided the endorsement statement is generated only if the provenance
+// statements are consistent and valid according to the input
+// VerificationOptions. If no provenances are provided, a provenance-less
+// endorsement is generated, if the input VerificationOptions does not contain
+// a reference provenance.
+func GenerateEndorsement(binaryName, binaryDigest string, verOpt *prover.VerificationOptions, validityDuration claims.ClaimValidity, provenances []ParsedProvenance) (*intoto.Statement, error) {
+	if (verOpt.GetSkipProvenanceVerification() == nil) && (verOpt.GetReferenceProvenance() == nil) {
```

Comment:
Just checking: Don't you get from `oneof` in the proto definition that exactly one of `verOpt` is not nil?